src/api/service/Auth.service.ts

```diff
@@ -76,4 +80,19 @@ export class AuthService {
       }
     }
   }
+
+  private static async getRole(role: number): Promise<{ id: string }> {
+    try {
+      return (await prisma.role.findFirst({
+        where: {
+          role: role,
+        },
+        select: {
+          id: true,
+        },
+      })) as { id: string };
+    } catch (e) {
+      throw e;
```

Comment:
meaningful names: `err: Error` or `error:  Error`

Reply:
So, I would like you to consider [this](https://kentcdodds.com/blog/get-a-catch-block-error-message-with-typescript)

Reply:
thank you for this! i did something like that but using the prisma errors directly:
```
      if (error instanceof Prisma.PrismaClientKnownRequestError) {
        logger.error(error);
        logger.info('Prisma error:', error);
        if (prismaErrorsCodes400.includes(error.code)) throw new PrismaError(error.message, 400);
        if (prismaErrorsCodes404.includes(error.code)) throw new PrismaError(error.message, 404);
        throw new PrismaError(error.message, 500);
      }
```
in every method directly called by the controller i managed the error with the  validation before mentioned... In your suggestion the method getRole or userAuthentication are called from it's respective services and there you will find the handle error. I would prefer just throw the error from the getRole or userAthentication and manage the error directly from the service an then at controller return the respective response

Reply:
that was a great implementation and I considered to complement as well, do not forget the error management every time as you can.